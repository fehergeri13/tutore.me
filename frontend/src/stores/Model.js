import {observable, action} from "mobx";

import Auth from "./Auth";
import Rest from "./Rest";

export default class Model {
    @observable.ref auth = new Auth(this);
    @observable.ref routingStore = undefined;
    @observable.ref rest = new Rest(this);

    @observable isMessageModalOpen = false;
    @observable messageUserId = undefined;
    @observable messageUsername = undefined;


    constructor(routingStore) {
        this.routingStore = routingStore;
    }

    @action openMessageModal(messageUserId, messageUsername) {
        this.isMessageModalOpen = true;
        this.messageUserId = messageUserId;
        this.messageUsername = messageUsername;
    }

    @action closeMessageModal() {
        this.isMessageModalOpen = false;
    }
}