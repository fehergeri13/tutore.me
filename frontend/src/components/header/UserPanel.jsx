import React from 'react';
import {inject, observer} from 'mobx-react';
import {Link} from "react-router-dom";

@inject('model')
@observer
export default class UserPanel extends React.Component {
    renderLoggedIn() {
        return <ul className="user">
            <li><Link to="/messages">Üzenetek</Link></li>
            <li><Link to="/user">Profilom</Link></li>
            <li><Link to="/logout">Kilépés</Link></li>
        </ul>
    }

    renderLoggedOut() {
        return <ul className="user">
            <li><Link to="/login">Bejelentkezés</Link></li>
        </ul>
    }

    render() {
        if (this.props.model.auth.isLoggedIn) {
            return this.renderLoggedIn();
        } else {
            return this.renderLoggedOut();
        }
    }
}