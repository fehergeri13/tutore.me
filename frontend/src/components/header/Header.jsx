import React from 'react';
import {inject, observer} from 'mobx-react';


import "./header.less"
import {Link} from "react-router-dom";
import UserPanel from "./UserPanel";

@inject('model')
@observer
export default class App extends React.Component {



    renderLoggedIn() {
        return <ul className="user">
            <li><Link to="/messages">Üzenetek</Link></li>
            <li><Link to="/user">Profilom</Link></li>
            <li><Link to="/logout">Kilépés</Link></li>
        </ul>
    }

    renderLoggedOut() {
        return <ul className="user">
            <li><Link to="/login">Bejelentkezés</Link></li>
        </ul>
    }

    renderUserPanel() {
        if (this.props.model.auth.isLoggedIn) {
            return this.renderLoggedIn();
        } else {
            return this.renderLoggedOut();
        }
    }


    render() {
        return <div className="header">
            <h1>tutore.me</h1>

            <input id="habmurger-toggle" type="checkbox"/>
            <label htmlFor="habmurger-toggle" className="hamburger">
                <i className="material-icons menu">menu</i>
                <i className="material-icons close">close</i>
            </label>

            <div className="links">
                <ul className="nav">
                    <li><Link to="/">Hirdetések</Link></li>
                    <li><Link to="/create">Új hirdetés</Link></li>
                </ul>

                <UserPanel/>
            </div>
        </div>;
    }
}
