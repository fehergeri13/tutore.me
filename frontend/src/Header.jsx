import React from 'react';


import "./header.less"

export default class App extends React.Component {
    render() {
        return <div className="header">
            <h1>tutore.me</h1>

            <div className="nav">
                <a href="#">Hirdetések</a>
                <a href="#">Új hirdetés</a>
            </div>

            <div className="user">
                <a href="#">Üzenetek</a>
                <a href="#">Profilom</a>
                <a href="#">Kilépés</a>
            </div>
        </div>;
    }
}
