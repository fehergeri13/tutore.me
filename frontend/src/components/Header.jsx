import React from "react";

import "./header.less";

export default class Header extends React.Component {
    render() {
        return <div className="header">
            <h1>TUTORE.ME - Korrepetálás hirdetés</h1>
            <p>
                Ingyenes korrepetálás hirdetés!
                <a href="#hirdetesek">Hirdetések</a>
                <a href="#feladas">Hirdetés feladás</a>
            </p>
        </div>
    }
}