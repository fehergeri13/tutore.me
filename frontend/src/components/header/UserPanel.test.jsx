import React from 'react';
import renderer from 'react-test-renderer';
import Header from "./Header";
import {MemoryRouter as Router} from "react-router-dom";
import {Provider} from 'mobx-react'
import Model from "../../stores/Model";
import UserPanel from "./UserPanel";

describe('Header', () => {
    it('renders correctly', () => {

        const model = new Model();

        const tree = renderer.create(
            <Provider model={model}><Router><UserPanel /></Router></Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders logged in', () => {

        const model = new Model();

        model.auth.isLoggedIn = true;

        const tree = renderer.create(
            <Provider model={model}><Router><UserPanel /></Router></Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders logged out', () => {

        const model = new Model();

        model.auth.isLoggedIn = false;

        const tree = renderer.create(
            <Provider model={model}><Router><UserPanel /></Router></Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
