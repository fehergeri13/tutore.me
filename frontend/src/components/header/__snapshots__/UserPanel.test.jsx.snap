// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<ul
  className="user"
>
  <li>
    <a
      href="/login"
      onClick={[Function]}
    >
      Bejelentkezés
    </a>
  </li>
</ul>
`;

exports[`Header renders logged in 1`] = `
<ul
  className="user"
>
  <li>
    <a
      href="/messages"
      onClick={[Function]}
    >
      Üzenetek
    </a>
  </li>
  <li>
    <a
      href="/user"
      onClick={[Function]}
    >
      Profilom
    </a>
  </li>
  <li>
    <a
      href="/logout"
      onClick={[Function]}
    >
      Kilépés
    </a>
  </li>
</ul>
`;

exports[`Header renders logged out 1`] = `
<ul
  className="user"
>
  <li>
    <a
      href="/login"
      onClick={[Function]}
    >
      Bejelentkezés
    </a>
  </li>
</ul>
`;
